### PR TITLE
Support sending multiple messages to a subscription

### PR DESCRIPTION
**Pull request by clawconduce**
I needed these changes to write multiple messages to a subscription connection.

### CODE REVIEW

**Comment by rafaeljusto** [comment]
Thanks Charles, I will take a better look this weekend and send some comments.

**Comment by clawconduce** [comment]
Thanks!  Also, this is what came to mind find first, so suggestions are welcome, or even completely different implementations as long as I can mock sending messages to subscriptions.

**Comment by clawconduce** [comment]
Was closing tabs and thought this would be useful:

https://github.com/garyburd/redigo/blob/master/redis/pubsub.go#L105

I believe the issue is that subscriptions constantly receive without sending commands between.  So I didn't see how that fit with the queue, which stored commands and responses.

File: redigomock.go [hunk 1]
```diff
@@ -18,14 +18,16 @@ type queueElement struct {
...
 type Conn struct {
-	ReceiveWait bool            // When set to true, Receive method will wait for a value in ReceiveNow channel to proceed, this is useful in a PubSub scenario
-	ReceiveNow  chan bool       // Used to lock Receive method to simulate a PubSub scenario
-	CloseMock   func() error    // Mock the redigo Close method
-	ErrMock     func() error    // Mock the redigo Err method
-	FlushMock   func() error    // Mock the redigo Flush method
-	commands    []*Cmd          // Slice that stores all registered commands for each connection
-	queue       []queueElement  // Slice that stores all queued commands for each connection
-	stats       map[cmdHash]int // Command calls counter
+	Subscription bool            // Enable this to support SubResponses
```
Comment (rafaeljusto) [inline redigomock.go]: Why do we need this flag? Isn't enough to have something in the SubResponses slice?

**Comment by rafaeljusto** [comment]
I created an example for your scenario here:
https://gist.github.com/rafaeljusto/55839bfc2325c1d05596

Could you please create test cases for your changes?

File: redigomock.go (hunk 1 above)
Comment (clawconduce) [inline redigomock.go]: Yes, I can remove the flag.

**Comment by clawconduce** [comment]
The scenario looks good, I'll add a test case as well.

**Comment by kaustavha** [comment]
+1, could use this feature.

**Comment by rafaeljusto** [comment]
@clawconduce Thanks!

**Pull request merged**